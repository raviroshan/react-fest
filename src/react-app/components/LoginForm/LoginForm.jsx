import React from 'react';

import './LoginForm.scss';

function LoginForm({ onSubmit }) {
  return (
    <div>
      <form
        className="container login"
        onSubmit={e => {
          e.preventDefault();
          const { username, password } = e.target.elements;
          onSubmit({
            username: username.value,
            password: password.value,
          });
        }}
      >
        <div className="form-group">
          <input
            data-testid="username"
            placeholder="Username..."
            name="username"
            className="form-control"
          />
        </div>
        <div className="form-group">

          <input
            data-testid="password"
            placeholder="Password..."
            type="password"
            name="password"
            className="form-control"
          />
        </div>
        <button type="submit" className="btn btn-primary">
          Submit
        </button>
      </form>
    </div>
  );
}

export default LoginForm;
