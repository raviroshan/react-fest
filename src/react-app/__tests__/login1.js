import React from 'react';
import { render, cleanup } from 'react-testing-library';
import LoginForm from '../components/LoginForm/LoginForm';

afterEach(cleanup);
test('calls onSubmit with the username and password when submitted', () => {
  const fakeUser = { username: 'test', password: 'test' };
  const handleSubmit = jest.fn();
  const { getByTestId, getByText } = render(<LoginForm onSubmit={handleSubmit} />);

  const usernameNode = getByTestId(/username/i);
  const passwordNode = getByTestId(/password/i);

  // Act
  usernameNode.value = fakeUser.username;
  passwordNode.value = fakeUser.password;
  getByText(/submit/i).click();

  // Assert
  expect(handleSubmit).toHaveBeenCalledTimes(1);
  expect(handleSubmit).toHaveBeenCalledWith(fakeUser);
});

test('snapshot', () => {
  const { container } = render(<LoginForm />);
  expect(container).toMatchSnapshot();
});
