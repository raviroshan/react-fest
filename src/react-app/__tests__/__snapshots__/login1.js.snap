// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div>
    <form
      class="container login"
    >
      <div
        class="form-group"
      >
        <input
          class="form-control"
          data-testid="username"
          name="username"
          placeholder="Username..."
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          data-testid="password"
          name="password"
          placeholder="Password..."
          type="password"
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
